targets/playground/Thymio2-natives.cpp

```diff
@@ -114,14 +113,28 @@ extern "C" void PlaygroundThymio2Native_leds_circle(AsebaVMState *vm)
 	ASEBA_UNUSED(l5);
 	ASEBA_UNUSED(l6);
 	ASEBA_UNUSED(l7);
-}
 
-static double paramToColor(sint16 param)
-{
-	return (param*5.46875+80) / 255.;
+	PlaygroundViewer* playgroundViewer(PlaygroundViewer::getInstance());
+	World* world(playgroundViewer->getWorld());
+	for (World::ObjectsIterator objectIt = world->objects.begin(); objectIt != world->objects.end(); ++objectIt)
+	{
+		AsebaThymio2 *thymio2 = dynamic_cast<AsebaThymio2*>(*objectIt);
+		if (thymio2 && (&(thymio2->vm) == vm))
+		{
+			thymio2->setLedIntensity(Thymio2::RING_0, l0/32.);
+			thymio2->setLedIntensity(Thymio2::RING_1, l1/32.);
+			thymio2->setLedIntensity(Thymio2::RING_2, l2/32.);
+			thymio2->setLedIntensity(Thymio2::RING_3, l3/32.);
+			thymio2->setLedIntensity(Thymio2::RING_4, l4/32.);
+			thymio2->setLedIntensity(Thymio2::RING_5, l5/32.);
+			thymio2->setLedIntensity(Thymio2::RING_6, l6/32.);
+			thymio2->setLedIntensity(Thymio2::RING_7, l7/32.);
+			return;
+		}
+	}
 }
 
-static double _fmax(double x, double y)
+static sint16 _imax(sint16 x, sint16 y)
```

Comment:
My fault for duplicating the max function in the first place, but now we should rather use std::max<sint16, sint16>() instead.